code/obj/item/pet_carrier.dm

```diff
@@ -0,0 +1,319 @@
+/// How many mobs can fit in the carrier by default.
+#define DEFAULT_MAX_CAPACITY 1
+/// Probability that a mob can break out of the carrier with *flip, from 0 to 100.
+#define BREAK_OUT_PROB 10
+/// Number of mobs named explicitly on examine() before switching to "there's a lot of mobs in here wow".
+#define EXPLICIT_NAME_LIMIT 3
+
+// Actionbar action defines
+#define RELEASE_MOB 0
+#define TRAP_MOB 1
+
+/**
+ * # Pet carriers.
+ * Code by DisturbHerb, icons by Azwald/Sunkiisu.
+ *
+ * A handheld item which can hold some mob instances inside with support for visually displaying its occupants with vis_contents.
+ * This was created without access to the pre-existing chicken carrier code so it could be pretty bad.
+ *
+ * The given mobs that a carrier is allowed to hold is determined by var/mob/allowed_mobs which should be overridden by any child types of the
+ * carrier. The maximum capacity is determined by var/carrier_max_capacity which can also be overridden.
+ *
+ * Whether a mob can break out or not is determined by var/can_break_out, and if FALSE, any occupant cannot escape the carrier on their own.
+ *
+ * There is support for spawning with an animal already inside by overriding var/mob/default_mob with a valid path.
+ *
+ * The icon for the carrier is constructed using a base where an overlay of the front of the carrier and two /obj/dummys are layered on top.
+ * These dummy objects are held in the carrier's vis_contents and each of them hold the mobs and the door of the carrier. The reason why the
+ * vis_contents of vis_contents_proxy hold the mobs rather than the carrier doing so directly is so that an alpha mask can prevent parts of the
+ * occupants from rendering outside of the inside of the carrier.
+ *
+ * For any mobs inside, the atmosphere they'll attempt to breathe from is taken from the current turf that the carrier occupies, even if the carrier
+ * is being held in someone's hand.
```

Comment:
Might suggest removing most of these comments, a lot of it is implementation details which would need to be rewritten when changing code so, and someone can read the doc comments anyways for understanding usage

Reply:
This is all DMDOC, which is documentation.